cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/AppLifecycleHttpHandlerTest.java

```diff
@@ -417,6 +426,95 @@ public void testDeployVersionedAndNonVersionedApp() throws Exception {
+    appDetails = getAppDetails(Id.Namespace.DEFAULT.getId(), "cfgAppWithVersion");
+    String appIdV2Version = appDetails.getAppVersion();
+    versions.add(appIdV2Version);
+
+    Assert.assertEquals(versions, getAppVersions(appId.getNamespace(), appId.getApplication()));
+
+    List<JsonObject> appList = getAppList(appId.getNamespace());
+    Set<String> receivedVersions = new HashSet<>();
+    for (JsonObject appRecord : appList) {
+      receivedVersions.add(appRecord.getAsJsonPrimitive("version").getAsString());
+    }
+    Assert.assertEquals(versions, receivedVersions);
+
+    appDetails = getAppDetails(appId.getNamespace(), appId.getApplication(), appIdVersion);
+    Assert.assertEquals(appIdVersion, appDetails.getAppVersion());
+    // when LCM is enabled a new app version is generated else -SNAPSHOT is overwritten
+    Assert.assertEquals(GSON.toJson(config), appDetails.getConfiguration());
+
+    // Get app info for the app with default versionId by versioned API
+    ApplicationDetail appDetailsDefault = getAppDetails(appId.getNamespace(), appId.getApplication(),
+                                                        appIdDefaultVersion);
+    Assert.assertEquals(appIdDefaultVersion, appDetailsDefault.getAppVersion());
+    // when LCM is enabled a new app version is generated else -SNAPSHOT is overwritten
+    Assert.assertEquals(GSON.toJson(configDefault), appDetailsDefault.getConfiguration());
+
+    // Get app info for the app with versionId "version_2" by versioned API
+    ApplicationDetail appDetailsV2 = getAppDetails(appId.getNamespace(), appId.getApplication(), appIdV2Version);
+    Assert.assertEquals(GSON.toJson(configV2), appDetailsV2.getConfiguration());
+    Assert.assertEquals(appIdV2Version, appDetailsV2.getAppVersion());
+
+    // Update app with default versionId by versioned API
+    ConfigTestApp.ConfigClass configDefault2 = new ConfigTestApp.ConfigClass("mno", "pqr");
+    AppRequest<ConfigTestApp.ConfigClass> requestDefault2 = new AppRequest<>(
+      new ArtifactSummary(artifactId.getName(), artifactId.getVersion().getVersion()), configDefault2);
+    Assert.assertEquals(200, deploy(appIdDefault.toEntityId(), requestDefault2).getResponseCode());
+
+    ApplicationDetail appDetailsDefault2 = getAppDetails(appIdDefault.getNamespaceId(), appIdDefault.getId());
+    Assert.assertEquals(GSON.toJson(configDefault2), appDetailsDefault2.getConfiguration());
+
+    // Get updated app info for the app with default versionId by versioned API
+    ApplicationDetail appDetailsDefault2WithVersion = getAppDetails(appIdDefault.getNamespaceId(),
+                                                                    appIdDefault.getId());
+    Assert.assertEquals(GSON.toJson(configDefault2), appDetailsDefault2WithVersion.getConfiguration());
+
+    Id.Application appIdDelete = Id.Application.from(appId.getNamespace(), appId.getApplication());
+    deleteApp(appIdDelete, 200);
+    enableLCMFlag(false);
```

Comment:
this should be set in an `@After` annotated cleanup method. Otherwise a failure in anything above will cause this line to be skipped, which will in turn cause a bunch of subsequent tests to fail.

Reply:
similar comment for other places this is done